# Exploring character data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
DATA_PATH = "../../../data/our_character2_data.csv"
pd.set_option("max_colwidth", None)

In [3]:
df_character = pd.read_csv(DATA_PATH)
print(df_character.info())
df_character.isna().sum() / len(df_character) * 100

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 497 entries, 0 to 496
Data columns (total 15 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Actor_name                       497 non-null    object 
 1   character_types                  497 non-null    object 
 2   character                        497 non-null    object 
 3   Freebase_character_ID            497 non-null    object 
 4   Freebase_actor_ID                497 non-null    object 
 5   Freebase_character/actor_map_ID  497 non-null    object 
 6   Actor_gender                     496 non-null    object 
 7   Actor_age_at_movie_release       479 non-null    float64
 8   Actor_date_of_birth              493 non-null    float64
 9   Actor_height_(in meters)         466 non-null    float64
 10  Actor_ethnicity_(Freebase ID)    332 non-null    object 
 11  unique_character_name            101 non-null    object 
 12  movie                 

Actor_name                          0.000000
character_types                     0.000000
character                           0.000000
Freebase_character_ID               0.000000
Freebase_actor_ID                   0.000000
Freebase_character/actor_map_ID     0.000000
Actor_gender                        0.201207
Actor_age_at_movie_release          3.621730
Actor_date_of_birth                 0.804829
Actor_height_(in meters)            6.237425
Actor_ethnicity_(Freebase ID)      33.199195
unique_character_name              79.678068
movie                               0.000000
Wikipedia_movie_ID                  0.000000
Freebase_movie_ID                   0.000000
dtype: float64

There are only 497 data points for the character type, which is insufficient for a reliable analysis. Therefore, we will not include it in our analysis as the results would lack accuracy due to the limited data.